crates/l3/appchain-core-contract-client/src/interfaces/core_contract.rs

```diff
@@ -0,0 +1,29 @@
+use color_eyre::Result;
+use appchain_utils::invoke_contract;
+use appchain_utils::LocalWalletSignerMiddleware;
+use starknet_core::types::{FieldElement, InvokeTransactionResult};
+
+pub struct CoreContract<'a> {
+    signer: &'a LocalWalletSignerMiddleware,
+    address: FieldElement,
+}
+
+impl<'a> CoreContract<'a> {
+    pub fn new(address: FieldElement, signer: &'a LocalWalletSignerMiddleware) -> Self {
+        Self { signer, address }
+    }
+
+    pub async fn update_state(
+        &self,
+        program_output: Vec<FieldElement>,
+        onchain_data_hash: FieldElement,
+        onchain_data_size: FieldElement,
+    ) -> Result<InvokeTransactionResult> {
+        let mut calldata = Vec::new();
```

Comment:
You can use `Vec::with_capacity(program_output.len() + 2)` instead

Reply:
done